src/firebaseRef.js

```diff
@@ -0,0 +1,49 @@
+(function() {
+  "use strict";
+
+  function FirebaseRef() {
+    this.urls = null;
+    this.registerUrl = function registerUrl(urlOrConfig) {
+
+      if (typeof urlOrConfig === 'string') {
+        this.urls = {};
+        this.urls.default = urlOrConfig;
+      }
+
+      if (angular.isObject(urlOrConfig)) {
+        this.urls = urlOrConfig;
+      }
+
+    };
+
+    this.$$checkUrls = function $$checkUrls(urlConfig) {
+      if (!urlConfig) {
+        return new Error('No Firebase URL registered. Use firebaseRefProvider.registerUrl() in the config phase. This is required if you are using $firebaseAuthService.');
+      }
+    };
+
+    this.$$createRefsFromUrlConfig = function $$createMultipleRefs(urlConfig) {
+      var error = this.$$checkUrls(urlConfig);
+      if (error) { throw error; }
+      var defaultUrl = urlConfig.default;
+      var defaultRef = new Firebase(defaultUrl);
+      delete urlConfig.default;
+      angular.forEach(urlConfig, function(value, key) {
+        if (!defaultRef.hasOwnProperty(key)) {
+          defaultRef[key] = new Firebase(value);
```

Comment:
Only methods directly declared on the Firebase ref will cause hasOwnProperty() to return true. It does not traverse the prototype chain:

> Every object descended from Object inherits the hasOwnProperty method. This method can be used to determine whether an object has the specified property as a direct property of that object; unlike the in operator, this method does not check down the object's prototype chain. [msdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty)

Thus, inherited or a non-enumerable property will return false and get overwritten. Overall, I'm not particularly happy with the idea of modifying the Firebase reference; other gotchas like this one may be waiting to trip us up.

Are there better alternatives? I know it's slightly less fun, but maybe just always placing the default at `firebaseRef.default` is the best answer? Perhaps just not using firebaseRef with multiple URLs is best? We can leave this up to the imagination of the small population of Edge Casians who will encounter it?